integration/simulation/l2_network.go

```diff
@@ -19,13 +19,30 @@ type L2NetworkCfg struct {
 	nodeAddresses    []string
 	avgLatency       uint64
 	avgBlockDuration uint64
+	p2p              p2p.P2P
 }
 
 // NewL2Network returns an instance of a configured L2 Network (no nodes)
-func NewL2Network(avgBlockDuration uint64, avgLatency uint64) *L2NetworkCfg {
+func NewL2Network(
+	nrNodes int,
+	avgBlockDuration uint64,
+	avgLatency uint64,
+	newP2P func(ourAddress string, allAddresses []string) p2p.P2P,
+) *L2NetworkCfg {
+	// We generate the P2P addresses for each node on the network.
+	var nodeAddresses []string
+	for i := 1; i <= nrNodes; i++ {
+		nodeAddresses = append(nodeAddresses, fmt.Sprintf("localhost:%d", P2P_START_PORT+i))
+	}
+
+	p2p := newP2P("localhost:11000", nodeAddresses)
```

Comment:
Also, my IDE is complaining that this hides the `p2p` package. I think that's valid, we should rename this variable.